internal/atlas/atlas.go

```diff
@@ -0,0 +1,214 @@
+		TxMode          string
+		Amount          uint
+	}
+	StatusParams struct {
+		DirURL          string
+		URL             string
+		RevisionsSchema string
+	}
+	LintParams struct {
+		DevURL  string
+		DirURL  string
+		Latest  uint
+		GitBase bool
+	}
+)
+
+func NewClient(name string) (*MigrateClient, error) {
+	path, err := execPath(name)
+	if err != nil {
+		return nil, err
+	}
+	return NewClientWithPath(path)
+}
+
+func NewClientWithPath(path string) (*MigrateClient, error) {
+	return &MigrateClient{path: path}, nil
+}
+
+func (c *MigrateClient) Apply(ctx context.Context, data *ApplyParams) (*ApplyReport, error) {
+	args := []string{
+		"migrate", "apply", "--log", "{{ json . }}",
+		"--url", data.URL,
+		"--dir", fmt.Sprintf("file://%s", data.DirURL),
+	}
+	if data.RevisionsSchema != "" {
+		args = append(args, "--revisions-schema", data.RevisionsSchema)
+	}
+	if data.BaselineVersion != "" {
+		args = append(args, "--baseline", data.BaselineVersion)
+	}
+	if data.TxMode != "" {
+		args = append(args, "--tx-mode", data.TxMode)
+	}
+	if data.Amount > 0 {
+		args = append(args, fmt.Sprintf("%d", data.Amount))
+	}
+	var report ApplyReport
+	if err := c.runCommand(ctx, args, &report); err != nil {
+		return nil, err
+	}
+	return &report, nil
+}
+
+func (c *MigrateClient) Lint(ctx context.Context, data *LintParams) (*SummaryReport, error) {
+	args := []string{
+		"migrate", "lint", "--log", "{{ json . }}",
+		"--dev-url", data.DevURL,
+		"--dir", fmt.Sprintf("file://%s", data.DirURL),
+	}
+	if data.Latest > 0 {
+		args = append(args, "--latest", fmt.Sprintf("%d", data.Latest))
+	}
+	if data.GitBase {
+		args = append(args, "--git-base")
+	}
+	var report SummaryReport
+	if err := c.runCommand(ctx, args, &report); err != nil {
+		return nil, err
+	}
+	return &report, nil
+}
+
+func (c *MigrateClient) Status(ctx context.Context, data *StatusParams) (*StatusReport, error) {
+	args := []string{
+		"migrate", "status", "--log", "{{ json . }}",
+		"--url", data.URL,
+		"--dir", fmt.Sprintf("file://%s", data.DirURL),
+	}
+	if data.RevisionsSchema != "" {
+		args = append(args, "--revisions-schema", data.RevisionsSchema)
+	}
+	var report StatusReport
+	if err := c.runCommand(ctx, args, &report); err != nil {
+		return nil, err
+	}
+	return &report, nil
+}
+
+func (c *MigrateClient) runCommand(ctx context.Context, args []string, report interface{}) error {
+	cmd := exec.CommandContext(ctx, c.path, args...)
+	cmd.Env = append(cmd.Env, "ATLAS_NO_UPDATE_NOTIFIER=1")
+	output, err := cmd.Output()
+	if err != nil {
+		exitErr, ok := err.(*exec.ExitError)
+		if !ok {
+			return err
+		}
+		if exitErr.Stderr != nil && len(exitErr.Stderr) > 0 {
+			return &cliError{
+				summary: string(exitErr.Stderr),
+				detail:  string(output),
+			}
+		}
+		if exitErr.ExitCode() != 1 || !json.Valid(output) {
+			return &cliError{
+				summary: "Atlas CLI",
+				detail:  string(output),
+			}
+		}
+		// When the exit code is 1, it means that the command
```

Comment:
No, when it is 0. 1 (technically >0) means there is an error.

Reply:
@masseelch: I need to read the output of `migrate lint` which always return non-zero for case of lint error.

Reply:
Are you sure? IIRC non-zero codes are returned only if we have diagnostics that should cause the build to crash. e.g, data-dependent changes still return diagnostics but return status 0.

Reply:
It depends on the configuration. See the second block here: https://atlasgo.io/lint/analyzers#destructive-changes